### PR TITLE
upgrade solana to 1.10.25

### LINKED ISSUE #1934: Update Solana dependencies to 1.10.*
⚠️ Don't use issues for support questions ("How does x feature work in Anchor? How to do x in Anchor?"). Use the discord instead https://discord.gg/GVRw4kHB (but don't ping maintainers). ⚠️

https://github.com/project-serum/anchor/blob/master/client/Cargo.toml#L18

breaks my app that depends on `solana-account-decoder = "1.10"`

```
error: failed to select a version for `solana-account-decoder`.
    ... required by package `myapp`
versions that meet the requirements `^1.10` are: 1.10.9, 1.10.20, 1.10.19, 1.10.18, 1.10.17, 1.10.16, 1.10.15, 1.10.14, 1.10.13, 1.10.12, 1.10.11, 1.10.10, 1.10.8, 1.10.7, 1.10.6, 1.10.5, 1.10.4, 1.10.3, 1.10.2, 1.10.1, 1.10.0


all possible versions conflict with previously selected packages.

  previously selected package `solana-account-decoder v1.9.13`
    ... which satisfies dependency `solana-account-decoder = "~1.9.13"` of package `anchor-client v0.24.0`
    ... which satisfies dependency `anchor-client = "^0.24"` of package `myapp`
```

### PR DESCRIPTION
**Pull request by dovahcrow**
~~Don't merge this pr for now. Still waiting for `mpl-token-metadata` 1.3.0 to be uploaded to crates.io.~~ Ready for review.

Fixes #1934

### CODE REVIEW

File: spl/Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ dex = ["serum_dex"]
 [dependencies]
 anchor-lang = { path = "../lang", version = "0.24.2", features = ["derive"] }
 serum_dex = { git = "https://github.com/project-serum/serum-dex", rev = "1be91f2", version = "0.4.0", features = ["no-entrypoint"], optional = true }
-solana-program = "~1.9.13"
-spl-token = { version = "~3.2.0", features = ["no-entrypoint"], optional = true }
+solana-program = "~1.10.25"
+spl-token = { version = "~3.3.0", features = ["no-entrypoint"], optional = true }
 spl-associated-token-account = { version = "~1.0.3", features = ["no-entrypoint"], optional = true }
-mpl-token-metadata = { version = "1.2.7", optional = true, features = ["no-entrypoint"] }
+mpl-token-metadata = {version = "1.3.0", git = "https://github.com/metaplex-foundation/metaplex-program-library", optional = true, features = ["no-entrypoint"]}
```
Comment (dovahcrow) [inline spl/Cargo.toml:27]: Waiting for metaplex uploading it to crates.io.

**Comment by gitmalong** [comment]
Why are versions specified with ~?

**Comment by dovahcrow** [comment]
@gitmalong ~1.10.25 forbids Solana 1.11.* since we don't know if 1.11 will have some breaking change.

**Comment by dovahcrow** [comment]
I'm closing this because the team already upgraded the Solana version to 1.10.29 in another commit (#2027)

**Pull request closed without merging**